libs/books/feature/src/lib/book-search/book-search.component.spec.ts

```diff
@@ -1,27 +1,80 @@
-import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { BOOKS } from './../constants/book-search-constants';
+import {
+  async,
+  ComponentFixture,
+  TestBed,
+  tick,
+  fakeAsync,
+  discardPeriodicTasks
+} from '@angular/core/testing';
 import { NoopAnimationsModule } from '@angular/platform-browser/animations';
 import { SharedTestingModule } from '@tmo/shared/testing';
-
 import { BooksFeatureModule } from '../books-feature.module';
 import { BookSearchComponent } from './book-search.component';
-
-describe('ProductsListComponent', () => {
+import { provideMockStore } from '@ngrx/store/testing';
+import { clearSearch, getAllBooks, searchBooks } from '@tmo/books/data-access';
+import { Store } from '@ngrx/store';
+describe('BookSearchComponent', () => {
   let component: BookSearchComponent;
   let fixture: ComponentFixture<BookSearchComponent>;
-
+  let store: Store;
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      imports: [BooksFeatureModule, NoopAnimationsModule, SharedTestingModule]
+      imports: [
+        BooksFeatureModule,
+        NoopAnimationsModule,
+        SharedTestingModule
+      ],
+      providers: [
+        Store,
+        provideMockStore({
+          initialState: {
+            books: {
+              entities: []
+            }
+          },
+          selectors: [{
+            selector: getAllBooks,
+            value: []
+          }]
+        }),
+      ],
     }).compileComponents();
   }));
-
   beforeEach(() => {
     fixture = TestBed.createComponent(BookSearchComponent);
     component = fixture.componentInstance;
     fixture.detectChanges();
+    store = fixture.debugElement.injector.get(Store);
   });
 
-  it('should create', () => {
-    expect(component).toBeDefined();
+  describe('ngOnInit()', () => {
+    beforeEach(() => {
+      spyOn(component.searchForm, 'valueChanges');
+      spyOn(store, 'dispatch');
+      spyOn(store, 'select').and.returnValue(BOOKS);
+    });
+
+    it('It should call searchBooks() when we have value for search term and actual time spent after entering search term is equal to 500ms', fakeAsync(() => {
+      component.ngOnInit();
+      component.searchForm.controls.term.setValue('Angular');
+      tick(500);
+      component.searchForm.valueChanges.subscribe(() => {
+        expect(store.dispatch).toHaveBeenCalledTimes(1);
+        expect(store.dispatch).toHaveBeenCalledWith(searchBooks({ term: 'Angular' }))
```

Comment:
You can keep only one statement here that should be fine.
You can keep this one and remove one from line no 63.

Reply:
Done